dask_expr/_backends.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import pandas as pd
+from dask.backends import CreationDispatch
+from dask.dataframe.backends import DataFrameBackendEntrypoint
+
+from dask_expr._dispatch import get_collection_type
+
+
+class DXCreationDispatch(CreationDispatch):
+    """Dask-expressions version of CreationDispatch"""
+
+    # TODO Remove after https://github.com/dask/dask/pull/10794
+    def dispatch(self, backend: str):
+        from dask.backends import detect_entrypoints
+
+        try:
+            impl = self._lookup[backend]
+        except KeyError:
+            entrypoints = detect_entrypoints(f"dask_expr.{self._module_name}.backends")
+            if backend in entrypoints:
+                return self.register_backend(backend, entrypoints[backend].load()())
+        else:
+            return impl
+        raise ValueError(f"No backend dispatch registered for {backend}")
+
+
+dataframe_creation_dispatch = DXCreationDispatch(
+    module_name="dataframe",
+    default="pandas",
+    entrypoint_class=DataFrameBackendEntrypoint,
+    name="dataframe_creation_dispatch",
+)
```

Comment:
Note that we want to introduce a **new** `CreationDispatch` object for dask-expr, because we want to distinguish "legacy" dask-dataframe creation functions from newer expression-based creation functions (to be discovered at the `"dask_expr.dataframe.backend"` entrypoint path instead of `"dask.dataframe.backend"`).

In both legacy and expr-based code, the same `"dask.dataframe.backend"` config value is used to specify the default backend.